paths_cli/compiling/engines.py

```diff
@@ -0,0 +1,54 @@
+from .topology import build_topology
+from .core import Builder
+from paths_cli.compiling.core import Parameter
+from .tools import custom_eval_int
+from paths_cli.compiling.plugins import EngineCompilerPlugin, CategoryPlugin
+
+try:
+    from simtk import openmm as mm
+except ImportError:
+    HAS_OPENMM = False
+else:
+    HAS_OPENMM = True
+
+def load_openmm_xml(filename):
+    if not HAS_OPENMM:  # -no-cov-
+        raise RuntimeError("OpenMM does not seem to be installed")
+
+    with open(filename, mode='r') as f:
+        obj = mm.XmlSerializer.deserialize(f.read())
+
+    return obj
+
+def openmm_options(dct):
+    n_steps_per_frame = dct.pop('n_steps_per_frame')
+    n_frames_max = dct.pop('n_frames_max')
+    options = {'n_steps_per_frame': n_steps_per_frame,
+               'n_frames_max': n_frames_max}
+    dct['options'] = options
+    return dct
+
+
+OPENMM_PARAMETERS = [
+    Parameter('topology', build_topology, json_type='string',
+              description=("File describing the topoplogy of this system; "
+                           "PDB recommended")),
+    Parameter('system', load_openmm_xml, json_type='string',
+              description="XML file with the OpenMM system"),
+    Parameter('integrator', load_openmm_xml, json_type='string',
+              description="XML file with the OpenMM integrator"),
+    Parameter('n_steps_per_frame', custom_eval_int,
+              description="number of MD steps per saved frame"),
+    Parameter("n_frames_max", custom_eval_int,
+              description=("maximum number of frames before aborting "
+                           "trajectory")),
```

Comment:
these parameters assume `custom_eval_int` returns a positive integer, which it isn't guaranteed to do

Reply:
Added `custom_eval_int_strict_pos` for this